uml/backend-software-system/input-listener.puml

```diff
@@ -0,0 +1,53 @@
+@startuml input-listener-console
+package "Console" {
+
+    abstract class IEventListener {
+        +handler: IEventHandler
+        +native_listen(): List<String>
+    }
+
+    class KeyboardListener {
+        +handler: IEventHandler
+        +native_listen(): List<String>
+    }
+
+    class MouseListener {
+        +handler: IEventHandler
+        +native_listen(): List<String>
+    }
+
+    KeyboardListener <|.. IEventListener
+    MouseListener <|.. IEventListener
+
+    abstract class IEventHandler {
+        -poolEvents: Vec<String>
+        +handleEvent(): void
+        +getPoolEvent(): Vec<String>
+    }
+
+    class KeyboardEventHandler {
+        -poolEvents: Vec<String>
+        +handleEvent(): void
+        +getPoolEvent(): Vec<String>
+    }
+
+    class MouseEventHandler {
+        -poolEvents: Vec<String>
+        +handleEvent(): void
+        +getPoolEvent(): Vec<String>
+    }
+
+    KeyboardEventHandler <|.. IEventHandler
+    MouseEventHandler <|.. IEventHandler
```

Comment:
Given that Mouse and Keyboard are inheriting from IEventHandler please update the relationships to generalization.

Reply:
solved